pkg/ingester/ingester_ingest_storage_test.go

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ingester
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/services"
+	"github.com/grafana/dskit/test"
+	"github.com/grafana/dskit/user"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kfake"
+	"github.com/twmb/franz-go/pkg/kmsg"
+	"go.uber.org/atomic"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+	"github.com/grafana/mimir/pkg/querier/api"
+	"github.com/grafana/mimir/pkg/storage/ingest"
+	util_test "github.com/grafana/mimir/pkg/util/test"
+	"github.com/grafana/mimir/pkg/util/testkafka"
+	"github.com/grafana/mimir/pkg/util/validation"
+)
+
+func TestIngester_QueryStream_IngestStorageReadConsistency(t *testing.T) {
```

Comment:
this test looks a lot like an end-to-end/integration test rather than a unit test. Do you want to move it to e2e instead? I'd even go as far as running the distributor and writing to it instead of writing to kafka. I can help with doing this change if you want in a follow-up PR

it would also remove the need to share ConsumerGroup, which IMO is an implementation detail of the storage/ingest package

Reply:
> this test looks a lot like an end-to-end/integration test rather than a unit test. Do you want to move it to e2e instead?

The way I see it is that an integration test shouldn't mock any component, while this test requires to mock the Kafka cluster, reason why I think it's appropriate for a unit test (also, we do pretty much the same kind of testing in most of the functions of `ingester_test.go`).

Reply:
> it would also remove the need to share ConsumerGroup, which IMO is an implementation detail of the storage/ingest package

I agree on this. I also was in doubt. For now I've hardcoded the consumer group name in `testkafka` package. We can worry about it when we'll need multiple ones: https://github.com/grafana/mimir/pull/7030/commits/c6af570f84f6ac6bc3e4316e8ba94c1a4284e99b

Reply:
> > this test looks a lot like an end-to-end/integration test rather than a unit test. Do you want to move it to e2e instead?
> 
> The way I see it is that an integration test shouldn't mock any component, while this test requires to mock the Kafka cluster, reason why I think it's appropriate for a unit test (also, we do pretty much the same kind of testing in most of the functions of `ingester_test.go`).

ok, I'm not feeling super strong about this. And testing is usually more conventions-based, so let's keep as-is